### DISCUSSION TITLE
Help about digging spherical porous in polycrystals to draw meshes

### DISCUSSION THREAD

**Discussion by mdlbyc** [neperfepx/neper#928] [Q&A] [answered]
I want to dig some spherical porous in polycrystals, but when I draw the mesh using neper's hexahedron technique, the surface of the spherical porous is very unsmooth, and if I draw the mesh using the tetrahedron technique, it leads to failure. The 3D meshing error message is:  [o] Writing file `failed_polys', in the file 'failed_polys', it only show '2'. The software version is 4.8.2. What is the best way to draw a mesh containing porous in a polycrystal using tetrahedral technique? Below are my codes of use of neper
$ neper -T -n 1000 -id 2 -domain "cube(1.0,1.0,1.0):translate(-0.5,-0.5,-0.5)" -transform "cut(sphere(0.0,0.0,0.0,0.4))"  -morpho voronoi -regularization 1  
$ neper -M n1000-id2.tess -elttype hex -rcl 0.5 -format inp           #the surface of the spherical porous is very unsmooth
$ neper -M n1000-id2.tess -elttype tet -format inp                    #The 3D meshing error message is:  [o] Writing file `failed_polys'

**Comment by rquey** [comment, marked as answer]
`-transform cut`  is an "experimental" feature and is not 100% reliable. The strategy is to try different `-id` values. It also helps to decrease the value of `-rsel`.

```
#!/bin/bash

id=1

while [ ! -f d.msh ]
do
  neper -T -n 1000 -id $id -domain "cube(1.0,1.0,1.0):translate(-0.5,-0.5,-0.5)" \
           -transform "cut(sphere(0.0,0.0,0.0,0.4))"  -morpho voronoi -regularization 1 -rsel 0.5 -o d
  neper -M d.tess -pl 10
  id=$[$id+1]
done

neper -V d.tess,d.msh -showelt "y>0" -print img

exit 0
```
![img](https://github.com/user-attachments/assets/3aca9563-871f-40eb-8fc9-a0427e900c5e)

**Reply by mdlbyc** [reply]
This is a great answer, thank you for your expertise!
